src/order/conditionally_complete_lattice.lean

```diff
@@ -769,3 +769,182 @@ instance (α : Type*) [conditionally_complete_linear_order α] :
   ..order_dual.decidable_linear_order α }
 
 end order_dual
+
+section with_top_bot
+
```

Comment:
Could you give a small docstring here explaining what you will do in this section, like
```lean
/-! ### Complete lattice structure on `with_top (with_bot α)`

We show how `with_top α` and ... inherit ... `complete_lattice`  ... Definition of  `Sup`...
-/
(exercise: fill the dots :)

Reply:
done